controller/orphan_controller.go

```diff
@@ -261,15 +261,27 @@ func (oc *OrphanController) reconcile(orphanName string) (err error) {
 		log.Infof("Orphan got new owner %v", oc.controllerID)
 	}
 
+	existingOrphan := orphan.DeepCopy()
+
 	if !orphan.DeletionTimestamp.IsZero() {
+		defer func() {
+			if reflect.DeepEqual(existingOrphan.Status, orphan.Status) {
+				return
+			}
+
+			if _, updateStatusErr := oc.ds.UpdateOrphanStatus(orphan); updateStatusErr != nil {
+				log.WithError(updateStatusErr).Errorf("Failed to update condition while cleaning up %v orphan %v", orphan.Spec.Type, orphan.Name)
+			}
+		}()
+
 		isCleanupComplete, err := oc.cleanupOrphanedResource(orphan)
 		if isCleanupComplete {
+			oc.eventRecorder.Eventf(orphan, corev1.EventTypeNormal, "OrphanCleanupComplete", "Orphan %v cleanup completed", orphan.Name)
```

Comment:
Should define the event in `constant/events.go`.

EventReasonOrphanCleanupCompleted = "OrphanCleanupCompleted"

```suggestion
			oc.eventRecorder.Eventf(orphan, corev1.EventTypeNormal, EventReasonOrphanCleanupCompleted, "Orphan %v cleanup completed", orphan.Name)
```